### PR TITLE
[LLT-4202] Make lookup return error in case of failure on forward

### PR DESCRIPTION
**Pull request by dfetti**
Make the forward error visible instead of just filling the response buffer with an empty response.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -180,26 +180,26 @@ jobs:
...
-        with:
-          path: target/bind
-          key: ${{ runner.os }}-bind-${{ hashFiles('**/justfile') }}
-          restore-keys: |
-            ${{ runner.os }}-bind-${{ hashFiles('**/justfile') }}
-            ${{ runner.os }}-bind
-
-      - name: just compatibility
-        run: just compatibility
+  # compatibility:
```
Comment (matszczygiel) [inline .github/workflows/test.yml:185]: Can you remind me why the compatibility tests fail?

File: tests/integration-tests/tests/catalog_tests.rs [hunk 2]
```diff
@@ -142,7 +142,7 @@ async fn test_catalog_lookup() {
     let question_req = Request::new(question_req, ([127, 0, 0, 1], 5553).into(), Protocol::Udp);
 
     let response_handler = TestResponseHandler::new();
-    catalog
+    let _ = catalog
```
Comment (matszczygiel) [inline tests/integration-tests/tests/catalog_tests.rs]: I think you should just `unwrap()` here since it's a test and you (probably) expect the call to be successful

File: .github/workflows/test.yml (hunk 1 above)
Comment (dfetti) [reply .github/workflows/test.yml:185]: I did not dig deeper after our talk. It was some check with bind9 which failed, but it should not impact us.

**Comment by matszczygiel** [comment]
+1

**Comment by mathiaspeters** [comment]
The commented out workflows should be commented why they're commented out and if there's isn't already there should be a ticket to fix it, or an issue opened upstream that can be linked in the aforementioned comment. It might make sense to add the comment both in the code and in the commit message

File: crates/server/src/store/sqlite/authority.rs [hunk 3]
```diff
@@ -452,8 +452,6 @@ impl SqliteAuthority {
     #[cfg_attr(docsrs, doc(cfg(feature = "dnssec")))]
     #[allow(clippy::blocks_in_if_conditions)]
     pub async fn authorize(&self, update_message: &MessageRequest) -> UpdateResult<()> {
-        use tracing::debug;
```
Comment (tomaszklak) [inline crates/server/src/store/sqlite/authority.rs:455]: Why is this removed? In telio we already convert tracing to logs: https://github.com/NordSecurity/libtelio/blob/9d5105a71645c6d20e951775035f6534fbe1d7e3/src/ffi/mod.rs#L170

**Comment by tomaszklak** [comment]
The commit message should also explain why we want to make lookup return error in case of failure on forward.

File: crates/server/src/store/sqlite/authority.rs (hunk 3 above)
Comment (dfetti) [reply crates/server/src/store/sqlite/authority.rs:455]: This is a leftover from the previous commit which I didn't catch until I runned the pipelines in this PR.
This is the [commit](https://github.com/NordSecurity/trust-dns/commit/887bba2af977666d45b78e8b644238892da8df67). It is just a cherry-pick of the original. I wrote it in its description.

**Review by tomaszklak — APPROVED** [review]
+1.0
